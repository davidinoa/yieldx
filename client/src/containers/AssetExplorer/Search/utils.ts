interface InGridFilter {
  filter: string | number;
  filterTo: number;
  filterType: string;
  type: string;
  dateFrom: string;
  dateTo: string;
  values: string[];
}

export interface AgGridFilters {
  [key: string]: any;
}

const getFilter = (filter: InGridFilter) => {
  /**
   *  The filter object from Ag Grid API. In the case of filtering `name` column.
   *  { filter: "Asset Name", filterType: "text", type: "contains"}
   */
  let searchFilter = {};
  const {
    filter: filterContent,
    filterType,
    type,
    filterTo,
    values,
    dateFrom,
    dateTo
  } = filter;
  if (filterType === "text") {
    switch (type) {
      case "contains":
        searchFilter = { containsText: filterContent };
        break;
      case "notContains":
        searchFilter = { doesNotContainText: filterContent };
        break;
      case "equals":
        searchFilter = { equalsText: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualText: filterContent };
        break;
      case "startsWith":
        searchFilter = { startsWithText: filterContent };
        break;
      case "endsWith":
        searchFilter = { endsWith: filterContent };
        break;
      default:
        searchFilter = { containsText: filterContent };
        break;
    }
  } else if (filterType === "number") {
    switch (type) {
      case "equals":
        searchFilter = { equalsNumber: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualNumber: filterContent };
        break;
      case "lessThan":
        searchFilter = { isLessThanNumber: filterContent };
        break;
      case "lessThanOrEqual":
        searchFilter = { isLessThanOrEqualToNumber: filterContent };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanNumber: filterContent };
        break;
      case "greaterThanOrEqual":
        searchFilter = { isGreaterThanOrEqualToNumber: filterContent };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToNumber: filterContent,
          isLessThanOrEqualToNumber: filterTo
        };
        break;
      default:
        searchFilter = { equalsNumber: filterContent };
        break;
    }
  } else if (filterType === "date") {
    const convertedDateFrom = new Date(dateFrom);
    const convertedDateTo = new Date(dateTo);
    switch (type) {
      case "equals":
        searchFilter = { equalsDate: convertedDateFrom };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanDate: convertedDateFrom };
        break;
      case "lessThan":
        searchFilter = { isLessThanDate: convertedDateFrom };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualDate: convertedDateFrom };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToDate: convertedDateFrom,
          isLessThanOrEqualToDate: convertedDateTo
        };
        break;
      default:
        searchFilter = { equalsDate: convertedDateFrom };
        break;
    }
  } else if (filterType === "set" && values) {
    // We may need to add `set` type Ag grid column filter params like cusip or isin
    // in the search request in the future.
    // searchFilter = { inText: values };
  }
  return searchFilter;
};

export const convertInGridFilters = (filters: AgGridFilters) => {
  const convertedFilters: { [index: string]: any } = {};
  Object.keys(filters).forEach(key => {
    const inGridFilter = filters[key];
    if (key === "minimumQuantity") key = "bond.minimumDenomination";
    convertedFilters[key] = getFilter(inGridFilter);
  });

  return convertedFilters;
};
