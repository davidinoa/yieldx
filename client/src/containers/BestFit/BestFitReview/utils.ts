import { getRatingValue } from "../../../util/ratings";
import YieldXPortfolio, {
  YieldXPosition
} from "../../../models/YieldXPortfolio";

export function getBondRatingsData(
  portfolio: YieldXPortfolio
): { name: string; value: number }[] {
  if (portfolio.getBondSleeve) {
    const ratingMarketValues: { [key: string]: number } = {};
    const bondPortfolio = portfolio.getBondSleeve();
    bondPortfolio.positions.forEach((position: YieldXPosition) => {
      const { asset, weight } = position;
      const rating = asset.rating ?? "NR";
      ratingMarketValues[rating] = (ratingMarketValues[rating] || 0) + weight;
    });
    return Object.entries(ratingMarketValues)
      .map(([rating, weight]) => {
        return {
          name: rating,
          value: weight
        };
      })
      .sort((a, b) => {
        return getRatingValue(a.name) - getRatingValue(b.name);
      });
  }
  return [];
}
