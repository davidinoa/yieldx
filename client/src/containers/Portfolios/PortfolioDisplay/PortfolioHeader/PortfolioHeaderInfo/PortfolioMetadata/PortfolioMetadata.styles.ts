import styled from "styled-components";

export const MetadataRow = styled.div`
  max-width: 45%;
  &.live-inpaas {
    max-width: 83%;
  }
  &.paper-inpaas {
    max-width: 66%;
  }
  border-top: 1px solid var(--border-new);
  .small-col {
    max-width: 160px;
  }
  .large-col {
    max-width: 200px;
  }
  .metadata-label {
    font-weight: bold;
    margin-bottom: 0.5rem;
    font-size: 11px;
    line-height: 13px;
    letter-spacing: 0.2em;
    text-transform: uppercase;
    color: var(--subtext-3);
  }

  .metadata-value {
    font-family: var(--font-secondary);
    font-size: 12px;
    line-height: 16px;
    a {
      display: inline-block;
      color: white;
    }
  }

  .goal-min-max {
    font-family: var(--font-secondary);
    font-size: 11px;
    line-height: 15px;
    color: var(--subtext-3);
  }

  .goal-separator {
    height: 10px;
    width: 1.5px;
    background: var(--border-new);
  }
`;
