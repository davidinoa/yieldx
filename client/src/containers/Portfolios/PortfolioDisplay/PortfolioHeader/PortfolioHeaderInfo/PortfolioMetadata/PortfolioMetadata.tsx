import React from "react";
import { ProposalRequest } from "@bondhouse/income-portfolios";
import Decimal from "util/Decimal";
import { roundToHundredth } from "util/roundToHundredth";
import getProposalTargetValues from "util/getProposalTargetValues";
import { MetadataRow } from "containers/Portfolios/PortfolioDisplay/PortfolioHeader/PortfolioHeaderInfo/PortfolioMetadata/PortfolioMetadata.styles";
import dayjs from "dayjs";
import { Link } from "react-router-dom";
import clsx from "clsx";

interface Props {
  proposalRequest?: ProposalRequest;
  savedPortfolioData?: {
    investorData?: {
      givenName: string;
      familyName: string;
      id: string;
    };
    createdAt: string;
  };
}

export default function PortfolioMetadata({
  proposalRequest,
  savedPortfolioData
}: Props) {
  const { investorData, createdAt } = savedPortfolioData || {};
  const targetValues = proposalRequest
    ? getProposalTargetValues(
        proposalRequest.goal,
        proposalRequest.investmentAmount
      )
    : null;
  const { targetRisk, targetYield, targetIncome } = targetValues || {
    targetYield: 0,
    targetRisk: 0,
    targetIncome: 0
  };
  const rowClass = clsx("row mt-4 pt-3", {
    "live-inpaas": Boolean(
      savedPortfolioData && savedPortfolioData.investorData && proposalRequest
    ),
    "paper-inpaas": Boolean(
      savedPortfolioData && !savedPortfolioData.investorData && proposalRequest
    )
  });
  return (
    <MetadataRow
      data-testid="PortfolioHeader-PortfolioMetadata"
      className={rowClass}
    >
      {proposalRequest !== undefined && (
        <>
          {!targetRisk ? (
            <>
              <div className="col d-flex flex-column small-col">
                <span className="metadata-label">target yield:</span>
                <span className="metadata-value">
                  {roundToHundredth(targetYield)}%
                </span>
              </div>
              <div className="col d-flex flex-column small-col">
                <span className="metadata-label">target income:</span>
                <span className="metadata-value">
                  ${Decimal.format(targetIncome)}
                </span>
              </div>
            </>
          ) : (
            <div className="col d-flex flex-column small-col">
              <span className="metadata-label">target risk:</span>
              <span className="metadata-value">
                {roundToHundredth(targetRisk)}%
              </span>
            </div>
          )}
          <div className="col d-flex flex-column small-col">
            <span className="metadata-label">allocation:</span>
            <div className="d-flex justify-content-between align-items-center mr-3">
              <div>
                <span className="metadata-value">
                  {proposalRequest?.constraints?.minimumWeight !== undefined
                    ? roundToHundredth(
                        proposalRequest?.constraints?.minimumWeight * 100
                      )
                    : 5}
                  %{" "}
                </span>
                <span className="goal-min-max">/min</span>
              </div>
              <div className="goal-separator" />
              <div>
                <span className="metadata-value">
                  {proposalRequest?.constraints?.maximumWeight !== undefined
                    ? roundToHundredth(
                        proposalRequest?.constraints?.maximumWeight * 100
                      )
                    : 25}
                  %{" "}
                </span>
                <span className="goal-min-max">/max</span>
              </div>
            </div>
          </div>
          {targetRisk !== undefined && !savedPortfolioData && (
            <div className="col " />
          )}
        </>
      )}
      {savedPortfolioData && createdAt && (
        <div className="col d-flex flex-column small-col">
          <span className="metadata-label">created at:</span>
          <span className="metadata-value">
            {dayjs.utc(createdAt).local().format("ddd MMM DD[ @]h:mma")}
          </span>
        </div>
      )}
      {investorData && (
        <div className="col d-flex flex-column large-col">
          <span className="metadata-label">investor account:</span>
          <span className="metadata-value">
            <Link to={`/investors/${investorData.id}`}>
              {`${investorData.givenName} ${investorData.familyName}`}
            </Link>
          </span>
        </div>
      )}
    </MetadataRow>
  );
}
