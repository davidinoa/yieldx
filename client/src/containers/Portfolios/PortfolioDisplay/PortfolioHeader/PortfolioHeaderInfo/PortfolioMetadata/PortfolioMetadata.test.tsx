import React from "react";
import { render } from "test/utils";
import PortfolioMetadata from "./PortfolioMetadata";

describe("PortfolioMetadata", () => {
  test("Displays targetYield as percentage and formats decimals when they exist", () => {
    const { asFragment, rerender } = render(
      <PortfolioMetadata
        targetYield={5}
        targetIncome={100}
        allocation={{
          minimumWeight: 0.05,
          maximumWeight: 0.25
        }}
      />
    );
    // Target yield should be shown as "5%"
    expect(asFragment()).toMatchSnapshot();

    rerender(
      <PortfolioMetadata
        targetYield={5.25}
        targetIncome={100}
        allocation={{
          minimumWeight: 0.05,
          maximumWeight: 0.25
        }}
      />
    );
    // Target yield should be shown as "5.25%"
    expect(asFragment()).toMatchSnapshot();

    rerender(
      <PortfolioMetadata
        targetYield={5.23333333}
        targetIncome={100}
        allocation={{
          minimumWeight: 0.05,
          maximumWeight: 0.25
        }}
      />
    );
    // Target yield should be shown as "5.23%"
    expect(asFragment()).toMatchSnapshot();
  });

  test("Target income should display as $ value and be formatted with commas and decimals", () => {
    const { asFragment, rerender } = render(
      <PortfolioMetadata
        targetYield={5}
        targetIncome={100}
        allocation={{
          minimumWeight: 0.05,
          maximumWeight: 0.25
        }}
      />
    );
    // Target income should be shown as "$100.00"
    expect(asFragment()).toMatchSnapshot();

    rerender(
      <PortfolioMetadata
        targetYield={5}
        targetIncome={10_250}
        allocation={{
          minimumWeight: 0.05,
          maximumWeight: 0.25
        }}
      />
    );
    // Target income should be shown as "$10,000.00"
    expect(asFragment()).toMatchSnapshot();

    rerender(
      <PortfolioMetadata
        targetYield={5}
        targetIncome={1_510_250.25}
        allocation={{
          minimumWeight: 0.05,
          maximumWeight: 0.25
        }}
      />
    );
    // Target income should be shown as "$1,510,250.25"
    expect(asFragment()).toMatchSnapshot();
  });

  test("Allocations should be multiplied by 100 and replace undefined values with defaults", () => {
    const { asFragment, rerender } = render(
      <PortfolioMetadata
        targetYield={5}
        targetIncome={100}
        allocation={{
          minimumWeight: 0.075,
          maximumWeight: 0.3
        }}
      />
    );
    // Min should be "7.5% Max should be "30%"
    expect(asFragment()).toMatchSnapshot();

    rerender(
      <PortfolioMetadata
        targetYield={5}
        targetIncome={10_250}
        allocation={{
          minimumWeight: 0.05
        }}
      />
    );
    // Min should be "5% Max should be "25%"
    expect(asFragment()).toMatchSnapshot();
  });
});
