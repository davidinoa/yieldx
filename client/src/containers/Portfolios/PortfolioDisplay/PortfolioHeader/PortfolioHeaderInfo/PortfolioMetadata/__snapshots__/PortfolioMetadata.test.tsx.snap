// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioMetadata Allocations should be multiplied by 100 and replace undefined values with defaults 1`] = `
<DocumentFragment>
  .c0 {
  max-width: 45%;
  border-top: 1px solid var(--border-new);
}

.c0.live-inpaas {
  max-width: 83%;
}

.c0.paper-inpaas {
  max-width: 66%;
}

.c0 .small-col {
  max-width: 160px;
}

.c0 .large-col {
  max-width: 200px;
}

.c0 .metadata-label {
  font-weight: bold;
  margin-bottom: 0.5rem;
  font-size: 11px;
  line-height: 13px;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--subtext-3);
}

.c0 .metadata-value {
  font-family: var(--font-secondary);
  font-size: 12px;
  line-height: 16px;
}

.c0 .metadata-value a {
  display: inline-block;
  color: white;
}

.c0 .goal-min-max {
  font-family: var(--font-secondary);
  font-size: 11px;
  line-height: 15px;
  color: var(--subtext-3);
}

.c0 .goal-separator {
  height: 10px;
  width: 1.5px;
  background: var(--border-new);
}

<div
    class="c0 row mt-4 pt-3"
    data-testid="PortfolioHeader-PortfolioMetadata"
  />
</DocumentFragment>
`;

exports[`PortfolioMetadata Allocations should be multiplied by 100 and replace undefined values with defaults 2`] = `
<DocumentFragment>
  .c0 {
  max-width: 45%;
  border-top: 1px solid var(--border-new);
}

.c0.live-inpaas {
  max-width: 83%;
}

.c0.paper-inpaas {
  max-width: 66%;
}

.c0 .small-col {
  max-width: 160px;
}

.c0 .large-col {
  max-width: 200px;
}

.c0 .metadata-label {
  font-weight: bold;
  margin-bottom: 0.5rem;
  font-size: 11px;
  line-height: 13px;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--subtext-3);
}

.c0 .metadata-value {
  font-family: var(--font-secondary);
  font-size: 12px;
  line-height: 16px;
}

.c0 .metadata-value a {
  display: inline-block;
  color: white;
}

.c0 .goal-min-max {
  font-family: var(--font-secondary);
  font-size: 11px;
  line-height: 15px;
  color: var(--subtext-3);
}

.c0 .goal-separator {
  height: 10px;
  width: 1.5px;
  background: var(--border-new);
}

<div
    class="c0 row mt-4 pt-3"
    data-testid="PortfolioHeader-PortfolioMetadata"
  />
</DocumentFragment>
`;

exports[`PortfolioMetadata Displays targetYield as percentage and formats decimals when they exist 1`] = `
<DocumentFragment>
  .c0 {
  max-width: 45%;
  border-top: 1px solid var(--border-new);
}

.c0.live-inpaas {
  max-width: 83%;
}

.c0.paper-inpaas {
  max-width: 66%;
}

.c0 .small-col {
  max-width: 160px;
}

.c0 .large-col {
  max-width: 200px;
}

.c0 .metadata-label {
  font-weight: bold;
  margin-bottom: 0.5rem;
  font-size: 11px;
  line-height: 13px;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--subtext-3);
}

.c0 .metadata-value {
  font-family: var(--font-secondary);
  font-size: 12px;
  line-height: 16px;
}

.c0 .metadata-value a {
  display: inline-block;
  color: white;
}

.c0 .goal-min-max {
  font-family: var(--font-secondary);
  font-size: 11px;
  line-height: 15px;
  color: var(--subtext-3);
}

.c0 .goal-separator {
  height: 10px;
  width: 1.5px;
  background: var(--border-new);
}

<div
    class="c0 row mt-4 pt-3"
    data-testid="PortfolioHeader-PortfolioMetadata"
  />
</DocumentFragment>
`;

exports[`PortfolioMetadata Displays targetYield as percentage and formats decimals when they exist 2`] = `
<DocumentFragment>
  .c0 {
  max-width: 45%;
  border-top: 1px solid var(--border-new);
}

.c0.live-inpaas {
  max-width: 83%;
}

.c0.paper-inpaas {
  max-width: 66%;
}

.c0 .small-col {
  max-width: 160px;
}

.c0 .large-col {
  max-width: 200px;
}

.c0 .metadata-label {
  font-weight: bold;
  margin-bottom: 0.5rem;
  font-size: 11px;
  line-height: 13px;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--subtext-3);
}

.c0 .metadata-value {
  font-family: var(--font-secondary);
  font-size: 12px;
  line-height: 16px;
}

.c0 .metadata-value a {
  display: inline-block;
  color: white;
}

.c0 .goal-min-max {
  font-family: var(--font-secondary);
  font-size: 11px;
  line-height: 15px;
  color: var(--subtext-3);
}

.c0 .goal-separator {
  height: 10px;
  width: 1.5px;
  background: var(--border-new);
}

<div
    class="c0 row mt-4 pt-3"
    data-testid="PortfolioHeader-PortfolioMetadata"
  />
</DocumentFragment>
`;

exports[`PortfolioMetadata Displays targetYield as percentage and formats decimals when they exist 3`] = `
<DocumentFragment>
  .c0 {
  max-width: 45%;
  border-top: 1px solid var(--border-new);
}

.c0.live-inpaas {
  max-width: 83%;
}

.c0.paper-inpaas {
  max-width: 66%;
}

.c0 .small-col {
  max-width: 160px;
}

.c0 .large-col {
  max-width: 200px;
}

.c0 .metadata-label {
  font-weight: bold;
  margin-bottom: 0.5rem;
  font-size: 11px;
  line-height: 13px;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--subtext-3);
}

.c0 .metadata-value {
  font-family: var(--font-secondary);
  font-size: 12px;
  line-height: 16px;
}

.c0 .metadata-value a {
  display: inline-block;
  color: white;
}

.c0 .goal-min-max {
  font-family: var(--font-secondary);
  font-size: 11px;
  line-height: 15px;
  color: var(--subtext-3);
}

.c0 .goal-separator {
  height: 10px;
  width: 1.5px;
  background: var(--border-new);
}

<div
    class="c0 row mt-4 pt-3"
    data-testid="PortfolioHeader-PortfolioMetadata"
  />
</DocumentFragment>
`;

exports[`PortfolioMetadata Target income should display as $ value and be formatted with commas and decimals 1`] = `
<DocumentFragment>
  .c0 {
  max-width: 45%;
  border-top: 1px solid var(--border-new);
}

.c0.live-inpaas {
  max-width: 83%;
}

.c0.paper-inpaas {
  max-width: 66%;
}

.c0 .small-col {
  max-width: 160px;
}

.c0 .large-col {
  max-width: 200px;
}

.c0 .metadata-label {
  font-weight: bold;
  margin-bottom: 0.5rem;
  font-size: 11px;
  line-height: 13px;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--subtext-3);
}

.c0 .metadata-value {
  font-family: var(--font-secondary);
  font-size: 12px;
  line-height: 16px;
}

.c0 .metadata-value a {
  display: inline-block;
  color: white;
}

.c0 .goal-min-max {
  font-family: var(--font-secondary);
  font-size: 11px;
  line-height: 15px;
  color: var(--subtext-3);
}

.c0 .goal-separator {
  height: 10px;
  width: 1.5px;
  background: var(--border-new);
}

<div
    class="c0 row mt-4 pt-3"
    data-testid="PortfolioHeader-PortfolioMetadata"
  />
</DocumentFragment>
`;

exports[`PortfolioMetadata Target income should display as $ value and be formatted with commas and decimals 2`] = `
<DocumentFragment>
  .c0 {
  max-width: 45%;
  border-top: 1px solid var(--border-new);
}

.c0.live-inpaas {
  max-width: 83%;
}

.c0.paper-inpaas {
  max-width: 66%;
}

.c0 .small-col {
  max-width: 160px;
}

.c0 .large-col {
  max-width: 200px;
}

.c0 .metadata-label {
  font-weight: bold;
  margin-bottom: 0.5rem;
  font-size: 11px;
  line-height: 13px;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--subtext-3);
}

.c0 .metadata-value {
  font-family: var(--font-secondary);
  font-size: 12px;
  line-height: 16px;
}

.c0 .metadata-value a {
  display: inline-block;
  color: white;
}

.c0 .goal-min-max {
  font-family: var(--font-secondary);
  font-size: 11px;
  line-height: 15px;
  color: var(--subtext-3);
}

.c0 .goal-separator {
  height: 10px;
  width: 1.5px;
  background: var(--border-new);
}

<div
    class="c0 row mt-4 pt-3"
    data-testid="PortfolioHeader-PortfolioMetadata"
  />
</DocumentFragment>
`;

exports[`PortfolioMetadata Target income should display as $ value and be formatted with commas and decimals 3`] = `
<DocumentFragment>
  .c0 {
  max-width: 45%;
  border-top: 1px solid var(--border-new);
}

.c0.live-inpaas {
  max-width: 83%;
}

.c0.paper-inpaas {
  max-width: 66%;
}

.c0 .small-col {
  max-width: 160px;
}

.c0 .large-col {
  max-width: 200px;
}

.c0 .metadata-label {
  font-weight: bold;
  margin-bottom: 0.5rem;
  font-size: 11px;
  line-height: 13px;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--subtext-3);
}

.c0 .metadata-value {
  font-family: var(--font-secondary);
  font-size: 12px;
  line-height: 16px;
}

.c0 .metadata-value a {
  display: inline-block;
  color: white;
}

.c0 .goal-min-max {
  font-family: var(--font-secondary);
  font-size: 11px;
  line-height: 15px;
  color: var(--subtext-3);
}

.c0 .goal-separator {
  height: 10px;
  width: 1.5px;
  background: var(--border-new);
}

<div
    class="c0 row mt-4 pt-3"
    data-testid="PortfolioHeader-PortfolioMetadata"
  />
</DocumentFragment>
`;
